### ISSUE TITLE
fixed BGRA in DX10

### ISSUE THREAD

**Issue by LifeH2O** [chunying/gaminganywhere#20]
`https://github.com/chunying/gaminganywhere/blob/fd3886ecd0b2a0d48c65158e549ff8608a9bb1c5/ga/server/event-driven/hook-function.cpp#L720`

`frame->pixelformat = PIX_FMT_BGRA;`

In this line pixelformat is being hardcoded which causes incorrect colors in Dragon Prophet in my case.

**Comment by chunying** [comment]
Have you tried to modify the value (PIX_FMT_BGRA) to other possible values, e.g., PIX_FMT_RGBA? Does it work for you?

**Comment by LifeH2O** [comment]
Yes. Setting it to RGBA corrected the colors. As it can vary for games, I am now getting this value from `filter-source-pixelformat = rgba` as it is done here
https://github.com/chunying/gaminganywhere/blob/fd3886ecd0b2a0d48c65158e549ff8608a9bb1c5/ga/module/filter-rgb2yuv/filter-rgb2yuv.cpp#L91
